hca/full_upload.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+import pprint
+import sys
+import uuid
+
+from io import open
+
+from .constants import Constants
+from .upload_to_cloud import upload_to_cloud
+
+
+def eprint(*args, **kwargs):
+    """Print to stderr."""
+    print(*args, file=sys.stderr, **kwargs)
+
+
+class FullUpload:
+    """Functions needed to fully add this functionality to the command line parser."""
+
+    CONSOLE_ARGUMENT = "upload"
+    FILE_OR_DIR_ARGNAME = "file_or_dir"
+    CREATOR_ID_ENVIRONMENT_VARIABLE = "creator_uid"
+    REPLICA_ARGNAME = "--replica"
+
+    @classmethod
+    def add_parser(cls, subparsers):
+        """Call from parser.py to create the parser."""
+        subparser = subparsers.add_parser(
+            cls.CONSOLE_ARGUMENT,
+            help="Upload a file or directory to the cloud, register each of the files, and bundle them together."
+        )
+
+        subparser.add_argument(
+            cls.FILE_OR_DIR_ARGNAME,
+            nargs="+",
+            help="Relative or direct path to folder with all bundle files."
+        )
+
+        subparser.add_argument(
+            cls.REPLICA_ARGNAME,
+            help="Which cloud to upload to first. One of 'aws', 'gc', or 'azure'.",
+            default="aws"
+        )
+        subparser.add_argument(
+            "--staging-bucket",
+            help="Bucket within replica to upload to.",
+            default="hca-dcp-staging-test")
+        return True
+
+    @classmethod
+    def _upload_files(cls, args):
+        filenames = []
+        files_to_upload = []
+        for path in args[cls.FILE_OR_DIR_ARGNAME]:
+
+            # If the path is a directory, add all files in the directory to the bundle
+            if os.path.isdir(path):
+                for filename in os.listdir(path):
+                    full_file_name = os.path.join(path, filename)
+                    filenames.append(filename)
+                    files_to_upload.append(open(full_file_name, "rb"))
+            else:  # It's a file
+                filenames.append(os.path.basename(path))
+                files_to_upload.append(open(path, "rb"))
+
+        # Print to stderr, upload the files to s3 and return a list of tuples: (filename, filekey)
+        eprint("Uploading the following keys to aws:")
+        for file in filenames:
+            eprint("\t", file)
+
+        uploaded_keys = upload_to_cloud(files_to_upload, args["staging_bucket"], args["replica"])
+
+        filename_key_list = list(zip(filenames, uploaded_keys))
```

Comment:
Yeah you're right. I think I only called it because I wanted to print and see what was in the zip at some point.